pkgs/build-support/vm/default.nix

```diff
@@ -334,8 +343,9 @@ rec {
      that allows you to boot into the VM and debug it interactively. */
 
   runInLinuxVM = drv: lib.overrideDerivation drv ({ memSize ? 512, QEMU_OPTS ? "", args, builder, ... }: {
-    requiredSystemFeatures = [ "kvm" ];
-    builder = "${bash}/bin/sh";
+    system = hostPkgs.system;
+    requiredSystemFeatures = if !hostPkgs.stdenv.isDarwin then [ "kvm" ] else [];
+    builder = "${hostPkgs.bash}/bin/sh";
     args = ["-e" (vmRunCommand qemuCommandLinux)];
```

Comment:
Changing `system` here is a hack. This would probably go away when using `pkgsCross` properly which currently doesn't work due to `qemu_kvm` (see error in PR).